internal/core/command/controller/messaging/external/external.go

```diff
@@ -0,0 +1,140 @@
+//
+// Copyright (C) 2022 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package external
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	bootstrapContainer "github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+
+	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
+
+	"github.com/edgexfoundry/edgex-go/internal/core/command/application"
+	"github.com/edgexfoundry/edgex-go/internal/core/command/container"
+)
+
+const (
+	RequestQueryTopic  = "RequestQueryTopic"
+	ResponseQueryTopic = "ResponseQueryTopic"
+)
+
+func OnConnectHandler(dic *di.Container) mqtt.OnConnectHandler {
+	return func(client mqtt.Client) {
+		lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+		config := container.ConfigurationFrom(dic.Get)
+		externalTopics := config.MessageQueue.External.Topics
+		qos := config.MessageQueue.External.QoS
+		retain := config.MessageQueue.External.Retain
+
+		requestQueryTopic := externalTopics[RequestQueryTopic]
+		responseQueryTopic := externalTopics[ResponseQueryTopic]
+		if token := client.Subscribe(requestQueryTopic, qos, commandQueryHandler(responseQueryTopic, qos, retain, dic)); token.Wait() && token.Error() != nil {
+			client.Disconnect(0)
```

Comment:
Should we be disconnecting here?

Reply:
I followed app-sdk's behavior:
https://github.com/edgexfoundry/app-functions-sdk-go/blob/main/internal/trigger/mqtt/mqtt.go#L157-L164

But considering we have command request topic to subscribe as well should we just log the error and not disconnect here ?

Reply:
K, IMO we should not disconnect here. We have the disconnect when exiting which should suffice. 
I will file an issue on the SDK to do the same.